lib/index.d.ts

```diff
@@ -9,14 +9,12 @@ declare function generate(options?: generate.Options, callback?: generate.Callba
 declare function generate(callback?: generate.Callback): generate.Generator;
 declare namespace generate {
 
-    type Callback = (err: Error | undefined, records: any | undefined) => void;
+    type Callback = (err?: Error, records?: any) => void;
 
     type MatcherFunc = (value: any) => boolean;
 
-    interface Generator extends stream.Readable {}
```

Comment:
AFAIC it's not needed to declare an interface for a class. You can declare `extends` directly on the class.